hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/metrics/RpcMetrics.java

```diff
@@ -97,10 +98,25 @@
   public String name() { return name; }
 
   public static RpcMetrics create(Server server, Configuration conf) {
+    setMetricTimeUnit(conf);
     RpcMetrics m = new RpcMetrics(server, conf);
     return DefaultMetricsSystem.instance().register(m.name, null, m);
   }
 
+  private static void setMetricTimeUnit(Configuration conf) {
```

Comment:
this can be error-prone because a process can have multiple Server binding at different ports at the same time.
So metricsTimeUnit is initialized multiple times for each Server creation. Hopefully they end up using the same metric unit. Otherwise this could hypothetically create confusion later on.

Reply:
Agree, let me make it this way. If server is instance of `RPC.Server`, we can make this change so that for anyone extending this Server from downstream application does not interfere with core-site config used in core Hadoop.

Reply:
Done